x/crosschain/migrations/v5/migrate_test.go

```diff
@@ -61,6 +63,76 @@ func TestMigrateStore(t *testing.T) {
 
 }
 
+func TestResetTestnetNonce(t *testing.T) {
+	t.Run("do not reset only mainnet nonce", func(t *testing.T) {
+		k, ctx, _, zk := keepertest.CrosschainKeeper(t)
+		mainnetChain := common.EthChain()
+		nonceLow := int64(1)
+		nonceHigh := int64(10)
+		tss := sample.Tss()
+		zk.ObserverKeeper.SetTSS(ctx, tss)
+		zk.ObserverKeeper.SetPendingNonces(ctx, observertypes.PendingNonces{
+			Tss:       tss.TssPubkey,
+			ChainId:   mainnetChain.ChainId,
+			NonceLow:  nonceLow,
+			NonceHigh: nonceHigh,
+		})
+		zk.ObserverKeeper.SetChainNonces(ctx, observertypes.ChainNonces{
+			Index:   mainnetChain.ChainName.String(),
+			ChainId: mainnetChain.ChainId,
+			Nonce:   uint64(nonceHigh),
+		})
+		err := v5.MigrateStore(ctx, k, k.GetObserverKeeper())
+		require.NoError(t, err)
+		pn, found := zk.ObserverKeeper.GetPendingNonces(ctx, tss.TssPubkey, mainnetChain.ChainId)
+		require.True(t, found)
+		require.Equal(t, nonceLow, pn.NonceLow)
+		require.Equal(t, nonceHigh, pn.NonceHigh)
+		cn, found := zk.ObserverKeeper.GetChainNonces(ctx, mainnetChain.ChainName.String())
+		require.True(t, found)
+		require.Equal(t, uint64(nonceHigh), cn.Nonce)
+	})
+	t.Run("reset only testnet nonce", func(t *testing.T) {
+		k, ctx, _, zk := keepertest.CrosschainKeeper(t)
+		testnetChains := []common.Chain{common.GoerliChain(), common.MumbaiChain(), common.BscTestnetChain(), common.BtcTestNetChain()}
+		nonceLow := int64(1)
+		nonceHigh := int64(10)
+		tss := sample.Tss()
+		zk.ObserverKeeper.SetTSS(ctx, tss)
+		for _, chain := range testnetChains {
+			zk.ObserverKeeper.SetPendingNonces(ctx, observertypes.PendingNonces{
+				Tss:       tss.TssPubkey,
+				ChainId:   chain.ChainId,
+				NonceLow:  nonceLow,
+				NonceHigh: nonceHigh,
+			})
+			zk.ObserverKeeper.SetChainNonces(ctx, observertypes.ChainNonces{
+				Index:   chain.ChainName.String(),
+				ChainId: chain.ChainId,
+				Nonce:   uint64(nonceHigh),
+			})
+		}
+		err := v5.MigrateStore(ctx, k, zk.ObserverKeeper)
+		require.NoError(t, err)
+		assertValues := map[common.Chain]int64{
+			common.GoerliChain():     226841,
+			common.MumbaiChain():     200599,
+			common.BscTestnetChain(): 110454,
+			common.BtcTestNetChain(): 4881,
+		}
+
+		for _, chain := range testnetChains {
+			pn, found := zk.ObserverKeeper.GetPendingNonces(ctx, tss.TssPubkey, chain.ChainId)
+			require.True(t, found)
+			require.Equal(t, assertValues[chain], pn.NonceHigh)
+			require.Equal(t, assertValues[chain], pn.NonceLow)
+			cn, found := zk.ObserverKeeper.GetChainNonces(ctx, chain.ChainName.String())
+			require.True(t, found)
+			require.Equal(t, uint64(assertValues[chain]), cn.Nonce)
+		}
```

Comment:
To make sure we can set values for mainnet chains and check these are not modified?